WebhookSender.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace SampleProject.Webhooks
+{
+    public class WebhookSender
+    {
+        private readonly HttpClient _httpClient;
+        private readonly IReportingService _reportingService;
+
+        public WebhookSender(HttpClient httpClient, IReportingService reportingService)
+        {
+            _httpClient = httpClient;
+        }
+
+        public async Task SendWebhook(WebhookConfiguration configuration, bool webhookIsActive)
+        {
+            var d = new WebhookDataStore("https://webhooks.checkout.com/datastore");
+            var additionalWebhookInfo = d.GetAdditionalInfo(configuration.WebhookId);
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, configuration.Url);
+            var webhookResponse = _httpClient.Send(httpRequest);
+
+            if (webhookResponse.StatusCode == HttpStatusCode.OK)
```

Comment:
use Switch instead.

Reply:
Consider whether you wish to update 'd' to "Success" in any other scenario than HttpStatusCode.OK.

If not, replace else ifs with else => await d.UpdateResult("failed")

Currently, if HttpStatusCode != NotFound/ServiceUnavailable/OK, status is not updated.